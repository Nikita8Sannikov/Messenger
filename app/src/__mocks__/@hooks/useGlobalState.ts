const mockUseAppSelector = jest.fn();
const mockUseAppDispatch = jest.fn();

export const useAppSelector = mockUseAppSelector;
export const useAppDispatch = mockUseAppDispatch;

export { mockUseAppSelector, mockUseAppDispatch };
