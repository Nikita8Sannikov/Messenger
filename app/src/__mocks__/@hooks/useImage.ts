const useImage = (src: string | null): string => {
	return src || "/assets/images/noPhoto.jpg";
};

export default useImage; 