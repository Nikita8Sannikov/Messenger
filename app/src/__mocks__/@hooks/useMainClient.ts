const mockMainApi = {
	getAnotherUser: jest.fn(),
	signIn: jest.fn(),
	signUp: jest.fn(),
	logout: jest.fn(),
	uploadAvatarAuth: jest.fn(),
	getFriendsNotification: jest.fn(),
	getMessageNotification: jest.fn(),
	openFile: jest.fn(),
};

const useMainClient = jest.fn().mockReturnValue({ 
	mainApi: mockMainApi,
	getProfile: jest.fn(),
	existProfile: jest.fn(),
	lifeTimeExpire: jest.fn(),
	removeProfile: jest.fn(),
	downloadLogFile: jest.fn(),
});

export default useMainClient;
export { mockMainApi }; 