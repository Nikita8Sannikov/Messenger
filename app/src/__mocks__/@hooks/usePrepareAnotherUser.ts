export interface IUsePrepareAnotherUser {
  isLoading: boolean;
  isExistProfile: boolean;
}

const mockUsePrepareAnotherUser = (): IUsePrepareAnotherUser => ({
	isLoading: false,
	isExistProfile: true,
});

export default mockUsePrepareAnotherUser; 