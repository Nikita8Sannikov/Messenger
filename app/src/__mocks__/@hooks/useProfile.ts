export default () => ({
    on: jest.fn(),
    off: jest.fn(),
    user: {
        avatarUrl: jest.fn(),
        fullName: jest.fn()
    },
})