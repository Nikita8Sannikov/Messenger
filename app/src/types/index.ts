// В данном файле хранятся общие абстрактные типы, применимые на всем уровне клиента

export type ValueOf<T> = T[keyof T];

export type TimeoutType = ReturnType<typeof setTimeout>;
