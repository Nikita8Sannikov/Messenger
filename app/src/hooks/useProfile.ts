import useMainClient from "@hooks/useMainClient";

// Возврат сущности "Профиль" конкретного пользователя по переданному идентификатору
export default function useProfile(userId?: string) {
	const mainClient = useMainClient();
	return mainClient.getProfile(userId);
}
