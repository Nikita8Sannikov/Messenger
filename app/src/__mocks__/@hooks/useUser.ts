import { mockUserService } from "./useProfile";

const useUser = jest.fn().mockReturnValue(mockUserService);

export default useUser; 