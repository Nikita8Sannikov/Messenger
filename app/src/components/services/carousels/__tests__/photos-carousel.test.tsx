import { render, screen, waitFor } from "@testing-library/react";

import PhotosCarousel from "@components/services/carousels/photos-carousel";
import { type IPhoto } from "@custom-types/models.types";
import { mockPhotosService } from "../../../../__mocks__/@hooks/useProfile";
import mockImagesCarouselStore from "../../../../__mocks__/@store/images-carousel";

describe("PhotosCarousel", () => {
	const mockPhoto: IPhoto = {
		id: "1",
		userId: "123",
		path: "/photo1.jpg",
		createdAt: "2024-01-01",
		size: "1024",
		extension: "jpg",
	};

	beforeEach(() => {
		jest.clearAllMocks();
		const store = mockImagesCarouselStore.getState();
		store.photoIndex = null;
		mockPhotosService.photos = [];
		mockPhotosService.count = 0;
	});

	it("should match snapshot when no photo is selected", () => {
		const { container } = render(<PhotosCarousel />);
		expect(container).toMatchSnapshot();
	});

	it("should match snapshot when photo is selected", async () => {
		mockPhotosService.photos = [ mockPhoto ];
		mockPhotosService.count = 1;

		const store = mockImagesCarouselStore.getState();
		store.photoIndex = 0;

		const { container } = render(<PhotosCarousel />);
		await screen.findByTestId("carousel");
		expect(container).toMatchSnapshot();
	});

	it("should not render when no photo is selected", () => {
		const { container } = render(<PhotosCarousel />);
		expect(container.firstChild).toBeNull();
	});

	it("should render carousel when photo is selected and exists in photos service", async () => {
		mockPhotosService.photos = [ mockPhoto ];
		mockPhotosService.count = 1;

		const store = mockImagesCarouselStore.getState();
		store.photoIndex = 0;

		render(<PhotosCarousel />);

		const carousel = await screen.findByTestId("carousel");
		expect(carousel).toBeTruthy();

		const image = screen.getByRole("img");
		expect(image.getAttribute("src")).toBe("/photo1.jpg");
		expect(image.getAttribute("alt")).toBe("1");
	});

	it("should load more photos when reaching threshold", async () => {
		mockPhotosService.photos = [
			mockPhoto,
			{
				...mockPhoto,
				id: "2",
				path: "/photo2.jpg",
				createdAt: "2024-01-02",
			},
		];
		mockPhotosService.count = 2;

		const store = mockImagesCarouselStore.getState();
		store.photoIndex = 0;

		render(<PhotosCarousel />);

		await screen.findByTestId("carousel");

		store.photoIndex = 1;

		await waitFor(() => {
			expect(mockPhotosService.getAllPhotos).toHaveBeenCalled();
		});
	});

	it("should not render when selected photo is not found", async () => {
		mockPhotosService.photos = [ mockPhoto ];
		mockPhotosService.count = 1;

		const store = mockImagesCarouselStore.getState();
		store.photoIndex = 999;

		const { container } = render(<PhotosCarousel />);
		expect(container.firstChild).toBeNull();
	});

	it("should not render when photos array is empty", async () => {
		mockPhotosService.photos = [];
		mockPhotosService.count = 0;

		const store = mockImagesCarouselStore.getState();
		store.photoIndex = 0;

		const { container } = render(<PhotosCarousel />);
		expect(container.firstChild).toBeNull();
	});
});
