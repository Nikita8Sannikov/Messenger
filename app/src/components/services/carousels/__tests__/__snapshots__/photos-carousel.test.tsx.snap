// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotosCarousel should match snapshot when no photo is selected 1`] = `<div />`;

exports[`PhotosCarousel should match snapshot when photo is selected 1`] = `
<div>
  <div
    class="carousel"
    data-testid="carousel"
  >
    <div
      class="carousel__photo"
    >
      <div
        class="photo"
      >
        <img
          alt="1"
          class="photo__image"
          data-testid="image"
          loading="eager"
          src="/photo1.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
